design/resource-constraints.md

```diff
@@ -0,0 +1,52 @@
+# Resource Constraints
+
+## Overview
+Resource Constraints allow the rook components to be in specific Kubernetes Quality of Service (QoS) classes. For this the components started by rook need to have resource requests and/or limits set depending on which class the component(s) should be in.
```

Comment:
Should also mention that proper set resources help scheduler to place pods on nodes that have sufficient resource for it and not starved, and can guarantee the pod to have at least that amount of cpu/memory share.

Reply:
https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/ would be a good link to link to.

Reply:
This isn't given for all components. This would not apply for OSDs and mons.

Mons currently don't get placed by Kubernetes but through an own implemented logic. This is a topic I wanted to talk about soon, to try improve the placement logic to use Kubernetes "native" placement methods and not a "full blown" placement logic on our side if possible.